src/internal/easee/thing.go

```diff
@@ -0,0 +1,87 @@
+package easee
+
+import (
+	"fmt"
+
+	"github.com/futurehomeno/cliffhanger/adapter"
+	"github.com/futurehomeno/cliffhanger/adapter/service/chargepoint"
+	"github.com/futurehomeno/cliffhanger/adapter/service/meterelec"
+	"github.com/futurehomeno/cliffhanger/adapter/thing"
+	"github.com/futurehomeno/fimpgo"
+	"github.com/futurehomeno/fimpgo/fimptype"
+)
+
+// Info is an object representing charger persisted information.
+type Info struct {
+	ChargerID string `json:"charger_id"`
+}
+
+type thingFactory struct {
+	client Client
+}
+
+func NewThingFactory(client Client) adapter.ThingFactory {
+	return &thingFactory{
+		client: client,
+	}
+}
+
+func (t *thingFactory) Create(mqtt *fimpgo.MqttTransport, adapter adapter.ExtendedAdapter, thingState adapter.ThingState) (adapter.Thing, error) {
+	info := &Info{}
+
+	if err := thingState.Info(info); err != nil {
+		return nil, fmt.Errorf("factory: failed to retrieve information: %w", err)
+	}
+
+	controller := NewController(t.client, info.ChargerID)
+	groups := []string{"ch_0"}
+
+	return thing.NewCarCharger(mqtt, &thing.CarChargerConfig{
+		InclusionReport: t.inclusionReport(info, thingState, groups),
+		ChargepointConfig: &chargepoint.Config{
+			Specification: t.chargepointSpecification(adapter, thingState, groups),
+			Controller:    controller,
+		},
+		MeterElecConfig: &meterelec.Config{
+			Specification: t.meterElecSpecification(adapter, thingState, groups),
+			Reporter:      controller,
+		},
+	}), nil
+}
+
+// inclusionReport returns inclusion report.
+func (t *thingFactory) inclusionReport(info *Info, thingState adapter.ThingState, groups []string) *fimptype.ThingInclusionReport {
+	return &fimptype.ThingInclusionReport{
+		Address:        thingState.Address(),
+		Alias:          "easee",
+		ProductHash:    "easee",
+		CommTechnology: "cloud",
+		ProductName:    "Easee Laderobot",
+		ManufacturerId: "easee",
+		DeviceId:       info.ChargerID,
+		PowerSource:    "ac",
+		Groups:         groups,
+	}
+}
+
+// meterElecSpecification returns specification for an electricity meter service.
+func (t *thingFactory) chargepointSpecification(adapter adapter.ExtendedAdapter, thingState adapter.ThingState, groups []string) *fimptype.Service {
+	return chargepoint.Specification(
+		adapter.Name(),
+		adapter.Address(),
+		thingState.Address(),
+		groups,
+		[]string{"available", "preparing", "charging", "paused", "finished", "unknown"}, // TODO: move to Cliffhanger
```

Comment:
States are not part of Cliffhanger unless they have a specific meaning, because they are - as of now - device specific.

The only "specific" meaning states for EV charger states are:
"charging" - because this state is used by max guard to assess whether EV charger can be disabled or not
"ready_to_charge" - Markus told me that this state is required for the FHX app to be able to initiate charging - not sure if this is true - but if it is true, then based on these states, you cannot start charging from FHX app now.
It is also possible that there was a miscommunication during our conversation.

Reply:
Asked Dawid about this issue and we decided to leave it as it is for know - we can come back here in the future when we'll gain more knowledge about state support between different vendors.